src/zcash/Note.hpp

```diff
@@ -0,0 +1,83 @@
+#ifndef _ZCNOTE_H_
+#define _ZCNOTE_H_
+
+#include "uint256.h"
+#include "Zcash.h"
+#include "Address.hpp"
+#include "NoteEncryption.hpp"
+
+namespace libzcash {
+
+class Note {
+public:
+    uint256 a_pk;
+    uint64_t value;
+    uint256 rho;
+    uint256 r;
+
+    Note(uint256 a_pk, uint64_t value, uint256 rho, uint256 r)
+        : a_pk(a_pk), value(value), rho(rho), r(r) {}
+
+    Note();
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action, int nType, int nVersion) {
+        READWRITE(a_pk);
```

Comment:
I don't think `Note` ever really needs to be serialized, just `NotePlaintext`, so I will remove this.